gammapy/background/make.py

```diff
@@ -0,0 +1,470 @@
+        ev_list_ds = EventListDataset.read(i_ev_file)
+        livetime = ev_list_ds.event_list.observation_live_time_duration
+        aeff_hdu = fits.open(i_aeff_file)['EFFECTIVE AREA']
+        # TODO: Gammapy needs a class that interprets IRF files!!!
+        if aeff_hdu.header.comments['LO_THRES'] == '[TeV]':
+            energy_threshold_unit = 'TeV'
+        energy_threshold = Quantity(aeff_hdu.header['LO_THRES'], energy_threshold_unit)
+        if DEBUG > 2:
+            print(' livetime {}'.format(livetime))
+            print(' energy threshold {}'.format(energy_threshold))
+
+        # fill events above energy threshold, correct livetime accordingly
+        data_set = ev_list_ds.event_list
+        data_set = data_set.select_energy((energy_threshold, energy_threshold*1.e6))
+
+        # construct events cube (energy, X, Y)
+        # TODO: UNITS ARE MISSING??!!! -> also in the fits tables!!!
+        #       in header there is EUNIT (TeV)!!!
+        #       hard coding the units for now !!!
+        ev_DETX = Angle(data_set['DETX'], 'degree')
+        ev_DETY = Angle(data_set['DETY'], 'degree')
+        ev_energy = Quantity(data_set['ENERGY'],
+                             data_set.meta['EUNIT'])
+        ev_cube_table = Table([ev_energy, ev_DETY, ev_DETX],
+                              names=('ENERGY', 'DETY', 'DETX'))
+        if DEBUG > 2:
+            print(ev_cube_table)
+
+        # fill events
+
+        # get correct data cube format for histogramdd
+        ev_cube_array = np.vstack([ev_cube_table['ENERGY'],
+                                   ev_cube_table['DETY'],
+                                   ev_cube_table['DETX']]).T
+
+        # fill data cube into histogramdd
+        ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
+                                                     [events_cube.energy_bins,
+                                                      events_cube.dety_bins,
+                                                      events_cube.detx_bins])
+        ev_cube_hist = Quantity(ev_cube_hist, '') # counts
+
+        # fill cube
+        events_cube.background += ev_cube_hist
+
+        # fill livetime for bins where E_max > E_thres
+        energy_max = events_cube.energy_bins[1:]
+        dummy_dety_max = np.zeros_like(events_cube.dety_bins[1:])
+        dummy_detx_max = np.zeros_like(events_cube.detx_bins[1:])
+        # define grid of max values (i.e. bin max values for each 3D bin)
+        energy_max, dummy_dety_max, dummy_detx_max = np.meshgrid(energy_max,
+                                                                 dummy_dety_max,
+                                                                 dummy_detx_max,
+                                                                 indexing='ij')
+        mask = energy_max > energy_threshold
+
+        # fill cube
+        livetime_cube.background += livetime*mask
+
+    return events_cube, livetime_cube
+
+
+def divide_bin_volume(cube):
```

Comment:
Such a function that just operates on one object is better located as a method on the object.
Any reason not to make this a `CubeBackgroundModel` member function?